lib/github.ts

```diff
@@ -148,11 +148,15 @@ async function fetchGitHubRepoContentFromDownloadUrl(
   downloadUrl: string
 ): Promise<void> {
   const resp = await fetchRepoFileByDownloadUrl(downloadUrl);
-  const fileContents =
-    typeof resp.data === 'string'
-      ? resp.data
-      : JSON.stringify(resp.data, null, 2);
-  fs.outputFileSync(dest, fileContents, 'utf8');
+  const contentType = resp.headers['content-type'];
+  let fileContents;
+
+  if (contentType.startsWith('text')) {
+    fileContents = Buffer.from(resp.data).toString('utf8');
+  } else {
+    fileContents = resp.data;
```

Comment:
iirc there were some scenarios where the data needs to be stringified and some when it doesn't

Reply:
In the `axios` call, I specified that the response type should always be `arrayBuffer`. But if the mime type begins with "text", then we extract the text string and encode it as 'utf8.' 

In the original code, if you received data whose mime type was not text, then you got a buffer instead of a string. Doing it in reverse is more reliable. It also works for both binary files (images) and text files.

Reply:
Awesome, thanks for clarifying!